controllers/cinderapi_controller.go

```diff
@@ -253,20 +253,19 @@ func (r *CinderAPIReconciler) reconcileDelete(ctx context.Context, instance *cin
 	// It's possible to get here before the endpoints have been set in the status, so check for this
 	if instance.Status.APIEndpoints != nil {
 		for _, ksSvc := range keystoneServices {
-			ksSvcSpec := keystonev1.KeystoneServiceSpec{
-				ServiceType:        ksSvc["type"],
-				ServiceName:        ksSvc["name"],
-				ServiceDescription: ksSvc["desc"],
-				Enabled:            true,
-				APIEndpoints:       instance.Status.APIEndpoints[ksSvc["name"]],
-				ServiceUser:        instance.Spec.ServiceUser,
-				Secret:             instance.Spec.Secret,
-				PasswordSelector:   instance.Spec.PasswordSelectors.Service,
+
+			ks, err := keystone.GetKeystoneServiceWithName(ctx, helper, ksSvc["name"], instance.Namespace)
+
+			if err != nil {
```

Comment:
@abays @Akrog Does this check like this make sense to you?
I guess I can't `if err != nil && k8s_errors.IsNotFound(err) {` because we return `err` or `nil` on the two if statements.
Not an expert here so I'd like to have your input!

Reply:
The way you now have it is the usual approach we've taken on past operators, so this looks good